fluentinterface/src/main/java/com/iluwatar/fluentinterface/fluentiterable/FluentIterable.java

```diff
@@ -0,0 +1,81 @@
+package com.iluwatar.fluentinterface.fluentiterable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * The FluentIterable is a more convenient implementation of the common iterable interface based
+ * on the fluent interface design pattern.
+ * This interface defines common operations, but
+ * doesn't aim to be complete. It was inspired by Guava's com.google.common.collect.FluentIterable.
+ * @param <TYPE> is the class of objects the iterable contains
+ */
+public interface FluentIterable<TYPE> extends Iterable<TYPE> {
+
+    /**
```

Comment:
Other way: "Filters the contents of `Iterable` using the given predicate, leaving only the ones which _satisfy_ the predicate."